tools/c7n_azure/c7n_azure/lookup.py

```diff
@@ -0,0 +1,74 @@
+# 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import jmespath
+
+
+class Lookup(object):
+    RESOURCE_SOURCE = 'resource'
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'oneOf': [
+                {RESOURCE_SOURCE: {'type': 'string'}}
+            ],
+            'default-value': {'oneOf': [
```

Comment:
I think right now everything will pass this schema because by default `additionalProperties` is True:
https://json-schema.org/understanding-json-schema/reference/object.html

Reply:
I don't understand. Not "everything" will pass this schema. What is an example of something that will pass that shouldn't?

Reply:
This has to have `'additionalProperties': False` otherwise will allow almost anything.

Reply:
`additionalProperties` is only on objects. 'default-value' is not allowing an object type though.

Reply:
Oh I see the typo for the `default-value` key. I can add it because it does add value for the typo, but if people pass additional values then it just ignores them, it's not like it breaks anything.  The current schema catches if you don't pass `resource` and if you pass a wrong value to `default-value`  - so I would say that's quite a bit different than "everything."

Reply:
You can pass `int`, `bool` or `object` to `resource` and it pass the validation:

```
        size:
            resource: true
            random-field: random
```

```
        size:
            resource: 
               random: random
            random-field: random
```

Reply:
Actually this schema is totally wrong and this helped me catch it! Will update the schema shortly.